src/node.js

```diff
@@ -388,17 +389,17 @@
       tickDone();
     }
 
+    function PObj(c) {
+      this.callback = c;
+      this.domain = process.domain || null;
+    }
+
     function nextTick(callback) {
       // on the way out, don't bother. it won't get fired anyway.
       if (process._exiting)
         return;
 
-      var obj = {
-        callback: callback,
-        domain: process.domain || null
-      };
-
-      nextTickQueue.push(obj);
+      nextTickQueue.push(new PObj(callback));
```

Comment:
@MayhemYDG escape analysis would trivially catch that - there is no benefit in doing this over the older version (doing `var obj = `) and then doing `.push` on that. 

I think one advantage of having a "class" here is that v8 currently produces better stack traces for it.